pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/predicate/jstl/JstlTransformContextAdapter.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.pulsar.functions.transforms.predicate.jstl;
+
+import com.datastax.oss.pulsar.functions.transforms.TransformContext;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.commons.collections4.Transformer;
+import org.apache.commons.collections4.map.LazyMap;
+import org.apache.pulsar.common.schema.KeyValue;
+import org.apache.pulsar.functions.api.Record;
+
+/**
+ * A java bean that adapts the underlying {@link TransformContext} to be ready for jstl expression
+ * language binding.
+ */
+public class JstlTransformContextAdapter {
+  private TransformContext transformContext;
+
+  /**
+   * A key transformer backing the lazy key map. It transforms top level key fields to either a
+   * value or another lazy map for nested evaluation.
+   */
+  private Transformer<String, Object> keyTransformer =
+      (fieldName) -> {
+        if (this.transformContext.getKeyObject() instanceof GenericRecord) {
+          GenericRecord genericRecord = (GenericRecord) this.transformContext.getKeyObject();
+          JstlPredicate.GenericRecordTransformer transformer =
+              new JstlPredicate.GenericRecordTransformer(genericRecord);
+          return transformer.transform(fieldName);
+        } else if (this.transformContext.getKeyObject() instanceof KeyValue) {
```

Comment:
this is not needed, we can drop this case, it is not possible in Pulsar

Reply:
removed as per the offline discussion